web/app/scripts/views/map/layers-controller.js

```diff
@@ -116,25 +121,47 @@
                     $rootScope.$broadcast('driver.views.map:filterdrawn', null);
                 });
 
+                ctl.map.on('zoomend', function() {
+                    MapState.setZoom(ctl.map.getZoom());
+                });
+
+                ctl.map.on('moveend', function() {
+                    MapState.setLocation(ctl.map.getCenter());
+                });
+
                 // TODO: Find a better way to ensure this doesn't happen until filterbar ready
                 // (without timeout, filterbar components aren't ready to listen yet)
                 // add filtered overlays
                 // this will trigger `driver.filterbar:changed` when complete
                 $timeout(FilterState.restoreFilters, 1000);
+            }).then(function() {
+                if (MapState.getLocation() && MapState.getZoom()) {
+                    ctl.map.setView(MapState.getLocation(), MapState.getZoom());
+                }
+
+                if (MapState.getFilterGeoJSON()) {
+                    var layer =  L.geoJson(MapState.getFilterGeoJSON());
+                    layer.setStyle(filterStyle);
+                    ctl.editLayers.addLayer(layer);
+                }
             });
         };
 
-        function filterShapeCreated(event) {
+        function filterShapeCreated(layer) {
             // TODO: is the shape type useful info?
             //var type = event.layerType;
-
-            var layer = event.layer;
-
             ctl.editLayers.clearLayers();
-            $rootScope.$broadcast('driver.views.map:filterdrawn', null);
+
+            layer.setStyle(filterStyle);
             ctl.editLayers.addLayer(layer);
+            $rootScope.$broadcast('driver.views.map:filterdrawn');
+
+            // Use GeoJSON instead of a normal layer - theres a strange bug likely stemming from
+            //  race conditions on the Leaflet Map object otherwise
```

Comment:
It could well be the Leaflet-Draw plugin - it was the layer generated by that library's event which simply would not be added to the map on pain of complaints about map._viewPort.min (this is from memory, so the exact var name is probably slightly different) being undefined.